# **Stepik "Поколение Python": курс для начинающих**

# ***Ввод-вывод данных***

## **Целочисленная арифметика. Часть 1**

### Три последовательных числа
<p>Напишите программу вывода на экран трех последовательно идущих чисел, каждое на отдельной строке. Первое число вводит пользователь, остальные числа вы должны сами вычислять в программе.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести три последовательно идущих числа в соответствии с условием задачи.</p>

In [2]:
x = input("")
print(x)
print(int(x)+1)
print(int(x)+2)

1
1
2
3


### Сумма трёх чисел
<p>Напишите программу, которая считывает три целых числа и выводит на экран их сумму. Каждое число записано в отдельной строке.</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести сумму введённых чисел.</p>

In [ ]:
a = input()
b = input()
c = input()
print(int(a)+int(b)+int(c))

1
2
3
6


### Куб 🧊
<p>Напишите программу, вычисляющую объём куба и площадь его полной поверхности по введенному значению длины ребра.</p>

<p>Формат входных данных<br>
На вход программе подается одно целое число – длина ребра.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст и числа в соответствии с условием задачи.</p>

<p>Примечание 1. Объем куба и площадь полной поверхности можно вычислить по формулам:<br>
V = a^3<br>

S = 6 * a^2</p>

<p>Примечание 2. Обратите внимание на то, что на текущем этапе обучения мы не знаем про оператор возведения в степень,
поэтому пользуемся определением степени числа – число умножается само на себя указанное количество раз. Например:<br>

𝑎^4 = a * a * a * a</p>

In [ ]:
a = int(input())
print("Объем = ", a*a*a)
print("Площадь полной поверхности = ", 6*a*a)

25
Объем =  15625
Площадь полной поверхности =  3750


### Значение функции
<p>Напишите программу, которая по введённым целым значениям 𝑎 и 𝑏 вычисляет значение следующего выражения:<br>

3(𝑎+𝑏)^3 + 275𝑏^2 − 127𝑎 − 41</p>

<p>Формат входных данных<br>
На вход программе подаются два целых числа (каждое на отдельной строке):
𝑎 и 𝑏.</p>

<p>Формат выходных данных<br>
Программа должна вывести значение функции по введённым числам 𝑎 и 𝑏.</p>

<p>Примечание. Обратите внимание на то, что на текущем этапе обучения мы не знаем про оператор возведения в степень,
поэтому пользуемся определением степени числа – число умножается само на себя указанное количество раз. Например:<br>

𝑎^4 = a * a * a * a</p>

In [ ]:
a = int(input())
b = int(input())
x = 3 * ((a + b) * (a + b) * (a + b)) + 275 * (b * b) - 127 * a - 41
print(x)

1
0
-165


### Следующее и предыдущее 🔢
<p>Напишите программу, которая считывает целое число и выводит для него на экран следующее и предыдущее целые числа в следующем формате:<br>

Следующее за числом <текущее число> число: <следующее число><br>
Для числа <текущее число> предыдущее число: <предыдущее число></p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [ ]:
a = int(input())
print("Следующее за числом", a, "число:", a + 1)
print(f"Для числа {a} предыдущее число: {a - 1}")

1
Следующее за числом 1 число: 2
Для числа 1 предыдущее число: 0


### Стоимость покупки 🛒
<p>Напишите программу, которая считает стоимость трёх компьютеров, состоящих из монитора, системного блока, клавиатуры и мыши.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых числа (каждое на отдельной строке):
стоимость монитора, стоимость системного блока, стоимость клавиатуры, стоимость мыши.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – стоимость покупки (трёх компьютеров).</p>

In [ ]:
mon = int(input())
sys = int(input())
key = int(input())
mouse = int(input())
print(3 * (mon + sys + key + mouse))

9900
55600
3999
2990
217467


### Арифметические операции
<p>Напишите программу, которая принимает на вход два числа 𝑎 и 𝑏, вычисляет сумму, разность и произведение для этих чисел и выводит текст в следующем формате:<br>

<число a> + <число b> = <сумма чисел a и b><br>
<число a> - <число b> = <разность чисел a и b><br>
<число a> * <число b> = <произведение чисел a и b></p>

<p>Формат входных данных<br>
На вход программе подаются два целых числа (каждое на отдельной строке):
𝑎 и 𝑏.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [ ]:
a = int(input())
b = int(input())
print(f"{a} + {b} = {a + b}")
print(f"{a} - {b} = {a - b}")
print(f"{a} * {b} = {a * b}")

2
7
2 + 7 = 9
2 - 7 = -5
2 * 7 = 14


### Арифметическая прогрессия
<p>Арифметической прогрессией называется последовательность чисел a1, a2, ..., an, каждое из которых, начиная с 𝑎2, получается из предыдущего путем прибавления к нему одного и того же постоянного числа 𝑑 (разность прогрессии).
Если известен первый член прогрессии (𝑎1) и её разность (𝑑), то 𝑛-ый член арифметической прогрессии находится по формуле:<br>
an = a1 + d(n − 1)</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа:<br>
𝑎1, 𝑑 и 𝑛, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести 𝑛-ый член арифметической прогрессии.</p>

<p>Примечание. Арифметической прогрессией является, например, следующая последовательность:<br>
−6,−3,0,3,6,9,12</p>

In [ ]:
a1 = int(input())
d = int(input())
n = int(input())
print(a1 + d * (n - 1))

-1
1
2
0


### Разделяй и властвуй
<p>Напишите программу, которая считывает целое положительное число 𝑥 и выводит на экран последовательность чисел x,2x,3x,4x и 5𝑥, разделённых тремя черточками.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое положительное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст согласно условию задачи.</p>

In [ ]:
a = int(input())
print(a, 2*a, 3*a, 4*a, 5 * a, sep = "---")

7
7---14---21---28---35


## **Целочисленная арифметика. Часть 2**

In [ ]:
"""Геометрическая прогрессия
Геометрической прогрессией называется последовательность чисел 𝑏1, 𝑏2, …, 𝑏𝑛, каждое из которых, начиная с 𝑏2,
получается из предыдущего умножением на одно и то же постоянное число 𝑞 (знаменатель прогрессии).

Если известен первый член прогрессии и ее знаменатель, то 𝑛-ый член геометрической прогрессии находится по формуле
𝑏𝑛 = 𝑏1 ⋅ 𝑞^(𝑛−1)

Входные данные
На вход программе подаются три целых числа:
𝑏1, 𝑞 и 𝑛 , каждое на отдельной строке.

Выходные данные
Программа должна вывести 𝑛-ый член геометрической прогрессии.
"""
b1 = int(input())
q = int(input())
n = int(input())
print(b1 * q ** (n - 1))

1
2
5
16


In [ ]:
"""Расстояние в метрах
Напишите программу, которая находит полное число метров по заданному числу сантиметров.

Формат входных данных
На вход программе подаётся натуральное число – количество сантиметров.

Формат выходных данных
Программа должна вывести одно число – полное число метров.
"""
b = int(input())
print(b // 100)

300
3


In [ ]:
"""Мандарины 🍊
𝑛 школьников делят k мандаринов поровну, неделящийся остаток остается в корзине.
Сколько целых мандаринов достанется каждому школьнику?
Сколько целых мандаринов останется в корзине?

Формат входных данных
На вход программе подаётся два целых числа: количество школьников и количество мандаринов, каждое на отдельной строке.

Формат выходных данных
Программа должна вывести два числа: количество мандаринов, которое достанется каждому школьнику, и количество мандаринов, которое останется в корзине, каждое на отдельной строке.
"""
a = int(input())
b = int(input())
print(b // a)
print(b % a)

12
6
0
6


In [ ]:
"""Сама неотвратимость
Безумный титан Танос собрал все 6 камней бесконечности и намеревается уничтожить половину населения Вселенной по щелчку пальцев.
При этом если население Вселенной является нечетным числом, то титан проявит милосердие и округлит количество выживших в большую сторону.
Помогите Мстителям подсчитать количество выживших.

Формат входных данных
На вход дается число целое
n – население Вселенной.

Формат выходных данных
Программа должна вывести одно число – количество выживших.
"""
a = int(input())
print(a - (a // 2))

99
50


In [ ]:
"""Пересчёт временного интервала ⌚
Напишите программу для пересчёта величины временного интервала, заданного в минутах, в величину, выраженную в часах и минутах в следующем формате:

<исходное кол-во минут> мин - это <полное кол-во часов> час <оставшееся кол-во минут> минут.
Формат входных данных
На вход программе подаётся целое число – количество минут.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
print(f"{a} мин - это {a // 60} час {a % 60} минут.")

50
50 мин - это 0 час 50 минут.


In [ ]:
"""Номер купе 🚉🌶️
В купейном вагоне имеется 9 купе с четырьмя местами для пассажиров в каждом.
Напишите программу, которая определяет номер купе, в котором находится место с заданным номером
(нумерация мест сквозная, начинается с 1).

Формат входных данных
На вход программе подаётся целое число – место с заданным номером в вагоне.

Формат выходных данных
Программа должна вывести одно число – номер купе, в котором находится указанное место.
"""
a = int(input())
print((a + 3) // 4)
# print(-((-a) // 4))

9
3


In [ ]:
"""Трёхзначное число
Напишите программу, которая рассчитывает сумму и произведение цифр положительного трёхзначного числа и выводит текст в следующем формате:

Сумма цифр = <сумма цифр>
Произведение цифр = <произведение цифр>
Формат входных данных
На вход программе подаётся положительное трёхзначное число.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
last = a % 10
mid = (a // 10) % 10
fir = a // 100
print("Сумма цифр =", last + mid + fir)
print("Произведение цифр =", last * mid * fir)

333
Сумма цифр = 9
Произведение цифр = 27


In [ ]:
"""Перестановка цифр
Дано трехзначное число abc, в котором все цифры различны.
Напишите программу, которая выводит шесть чисел, образованных при перестановке цифр заданного числа.

Формат входных данных
На вход программе подаётся положительное трёхзначное целое число, все цифры которого различны.

Формат выходных данных
Программа должна вывести шесть чисел, образованных при перестановке цифр заданного числа, в следующем порядке (каждое на новой строке):
𝑎𝑏𝑐, 𝑎𝑐𝑏, 𝑏𝑎𝑐, 𝑏𝑐𝑎, 𝑐𝑎𝑏, 𝑐𝑏𝑎
"""
num = int(input())
c = num % 10
b = (num // 10) % 10
a = num // 100
print(a, b, c, sep = '')
print(a, c, b, sep = '')
print(b, a, c, sep = '')
print(b, c, a, sep = '')
print(c, a, b, sep = '')
print(c, b, a, sep = '')

123
123
132
213
231
312
321


In [ ]:
"""Четырёхзначное число
Напишите программу для нахождения цифр четырёхзначного числа.

Формат входных данных
На вход программе подаётся положительное четырёхзначное целое число.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
last = a % 10
mid2 = (a // 10) % 10
mid1 = (a // 100) % 10
fir = a // 1000
print("Цифра в позиции тысяч равна", fir)
print("Цифра в позиции сотен равна", mid1)
print("Цифра в позиции десятков равна", mid2)
print("Цифра в позиции единиц равна", last)

1234
Цифра в позиции тысяч равна 1
Цифра в позиции сотен равна 2
Цифра в позиции десятков равна 3
Цифра в позиции единиц равна 4


# ***Итоговая работа на ввод-вывод данных***

### Прямоугольник
<p>Напишите программу, которая выводит прямоугольник, по периметру состоящий из звёздочек (*).</p>

<p>Примечание. Высота и ширина прямоугольника равны 4 и 17 звёздочкам соответственно.</p>
<p>Тестовые данные 🟢</p>
<p>Sample Input:<br>
</p>

<p>Sample Output:<br>

```
*****************
*               *
*               *
*****************
```

</p>

In [ ]:
print("*****************")
print("*               *")
print("*               *")
print("*****************")

*****************
*               *
*               *
*****************


### Квадрат суммы 🆚 Сумма квадратов
<p>Напишите программу, которая считывает два целых числа a и b и выводит на экран квадрат суммы (a+b)^2 и сумму квадратов (𝑎^2) + (b^2) этих чисел в следующем формате:</p>

<p>Напишите программу, которая считывает четыре целых положительных числа a,b,c и d и выводит на экран значение выражения<br>
(𝑎^𝑏) + (𝑐^𝑑).</p>

<p>Квадрат суммы a и b равен <квадрат суммы a и b><br>
Сумма квадратов a и b равна <сумма квадратов a и b></p>
<p>Формат входных данных<br>
На вход программе подаются два целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием.</p>

In [ ]:
a, b = int(input()), int(input())
print(f"Квадрат суммы {a} и {b} равен {(a + b) ** 2}")
print(f"Сумма квадратов {a} и {b} равна {a ** 2 + b ** 2}")

3
2
Квадрат суммы 3 и 2 равен 25
Сумма квадратов 3 и 2 равна 13


### Большое число
<p>Как известно, целые числа в языке Python не имеют ограничений, которые встречаются в других языках программирования.</p>

<p>Напишите программу, которая считывает четыре целых положительных числа a,b,c и d и выводит на экран значение выражения<br>
(𝑎^𝑏) + (𝑐^𝑑).</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых положительных числа a,b,c и 𝑑, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести значение (𝑎^𝑏) + (𝑐^𝑑).</p>

In [ ]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
print(a ** b + c ** d)

9
29
7
27
4710194409608608369201743232


### Размножение n-ок
<p>Напишите программу, которая считывает целое положительное число 𝑛, 𝑛 ∈ [1;9] и выводит значение числа n + nn + nnn.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое положительное число 𝑛, 𝑛 ∈ [1;9].</p>

<p>Формат выходных данных<br>
Программа должна вывести число n + nn + nnn.</p>

<p>Примечание. Для первого теста<br>
1+11+111=123.</p>

In [ ]:
a = input()
b = a + a
c = a + a + a
print(int(a) + int(b) + int(c))

1
123


# ***Условный оператор***

## ***Выбор из двух***

### Пароль 🔒
<p>При регистрации на сайтах требуется вводить пароль дважды. Это сделано для безопасности, поскольку такой подход уменьшает возможность неверного ввода пароля.<br>

Напишите программу, которая сравнивает пароль и его подтверждение. Если они совпадают, то программа выводит: «Пароль принят», иначе: «Пароль не принят».</p>

<p>Формат входных данных<br>
На вход программе подаются две строки.</p>

<p>Формат выходных данных<br>
Программа должна вывести одну строку в соответствии с условием задачи.</p>

In [3]:
a, b = input(), input()
if a == b:
    print("Пароль принят")
else:
    print("Пароль не принят")


qwerty
qwerty
Пароль принят


### Четное или нечетное?
<p>Напишите программу, которая определяет, является число четным или нечетным.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести «Четное» (без кавычек), если число четное, или «Нечетное» (без кавычек) в противном случае.</p>

In [5]:
a = int(input())
if a % 2 == 0:
    print("Четное")
else:
    print("Нечетное")

5
Нечетное


### Соотношение
<p>Напишите программу, которая проверяет, что для заданного четырехзначного числа выполняется следующее соотношение: сумма первой и последней цифр равна разности второй и третьей цифр.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое положительное четырёхзначное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести «ДА», если соотношение выполняется, и «НЕТ» — если не выполняется.</p>

In [9]:
a = int(input())
last = a % 10
mid2 = (a // 10) % 10
mid1 = (a // 100) % 10
fir = a // 1000
if fir + last == mid1 - mid2:
    print("ДА")
else:
    print("НЕТ")

1614
ДА


### Роскомнадзор 🔞
<p>Напишите программу, которая определяет, разрешён ли пользователю доступ к интернет-ресурсу или нет.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число – возраст пользователя.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст «Доступ разрешен», если возраст пользователя не менее 18 лет, или «Доступ запрещен» в противном случае.</p>

In [11]:
a = int(input())
if a < 18:
    print("Доступ запрещен")
else:
    print("Доступ разрешен")

11
Доступ запрещен


### Арифметическая прогрессия
<p>Напишите программу, которая определяет, являются ли три заданных числа (в указанном порядке) последовательными членами арифметической прогрессии.</p>

<p>Формат входных данных<br>
На вход программе подаются три числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести «YES» или «NO» (без кавычек) в соответствии с условием задачи.</p>

In [14]:
a, b, c = int(input()), int(input()), int(input())
if (a + c) / 2 * 3 != a + b + c:
    print("NO")
else:
    print("YES")

2
4
7
NO


### Наименьшее из двух чисел
<p>Напишите программу, которая определяет наименьшее из двух чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются два различных целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести наименьшее из двух чисел.</p>

In [15]:
a, b = int(input()), int(input())
if a < b:
    print(a)
else:
    print(b)

44
33
33


### Наименьшее из четырёх чисел 🌶️
<p>Напишите программу, которая определяет наименьшее из четырёх чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести наименьшее из четырёх чисел.</p>

<p>Примечание. Учитывайте, что минимальные числа могут дублироваться (смотрите тест №5). В таком случае необходимо вывести только одно из них.</p>

In [24]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
# print(min(a, b, c, d))

if a < b:
    b = a
else:
    a = b

if c < d:
    d = c
else:
    c = d

if a < d:
    d = a
else:
    a = d
print(a)

100
200
4
66
4


### Возрастная группа 👧🏻
<p>Напишите программу, которая по введённому возрасту пользователя сообщает, к какой возрастной группе он относится:<br>
до 13 (включительно) – детство;<br>
от 14 до 24 (включительно) – молодость;<br>
от 25 до 59 (включительно) – зрелость;<br>
от 60 (включительно) – старость.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число – возраст пользователя.</p>

<p>Формат выходных данных<br>
Программа должна вывести название возрастной группы.</p>

In [25]:
a = int(input())
if a <= 13:
    print("детство")
elif 13 < a <= 24:
    print("молодость")
elif 24 < a <= 59:
    print("зрелость")
else:
    print("старость")

# print("старость" if n>59 else "зрелость" if n>24 else "молодость" if n>13 else "детство")

14
молодость


### Только + 🌶️
<p>Напишите программу, которая считывает три числа и подсчитывает сумму только положительных чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа, каждое на новой строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – сумму положительных чисел.</p>

<p>Примечание. Если положительных чисел нет, то следует вывести 0.</p>

In [29]:
a, b, c = int(input()), int(input()), int(input())

if a <= 0:
    a = 0
if b <= 0:
    b = 0
if c <= 0:
    c = 0
print(a + b + c)

# print((a if a>0 else 0) + (b if b>0 else 0) + (c if c>0 else 0))

4
-22
1
5


## ***Логические операции***

### Принадлежность 1
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанному промежутку. </p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/9bd3a69a-d538-41a2-aefb-e76386c653f9/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [3]:
a = int(input())
if a > -1 and a < 17:
    print("Принадлежит")
else:
    print("Не принадлежит")

Принадлежит


### Принадлежность 2
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанным промежуткам.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/5d994476-13fb-40aa-a529-4fd12c9e424a/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [6]:
a = int(input())
if a <= -3 or a >= 7:
    print("Принадлежит")
else:
    print("Не принадлежит")

Принадлежит


### Принадлежность 3
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанным промежуткам.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/4659a972-d8f5-4186-90ed-38dfdacf17fb/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [10]:
a = int(input())
if -30 < a <= -2 or 7 < a <= 25:
    print("Принадлежит")
else:
    print("Не принадлежит")

Не принадлежит


### Красивое число 🌶️
<p>Назовем число красивым, если оно является четырехзначным и делится нацело на 7 или на 17. Напишите программу, определяющую, является ли введённое число красивым. Программа должна вывести «YES», если число является красивым, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаётся натуральное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [16]:
a = int(input())
if 1000 <= a < 10000 and (a % 7 == 0 or a % 17 == 0):
    print("YES")
else:
    print("NO")

NO


### Неравенство треугольника
<p>Напишите программу, которая принимает три положительных числа и определяет, существует ли невырожденный треугольник с такими сторонами.</p>

<p>Формат входных данных<br>
На вход программе подаётся три положительных целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести «YES» или «NO» в соответствии с условием задачи.</p>

<p>Примечание. Треугольник существует, если выполняется неравенство треугольника:<br>
a+b>c<br>
a+c>b<br>
b+c>a<br>
где a,b,c - стороны треугольника.</p>

In [18]:
a, b, c = int(input()), int(input()), int(input())
if a + b > c and a + c > b and b + c > a:
    print("YES")
else:
    print("NO")

YES


### Високосный год
<p>Напишите программу, которая определяет, является ли год с данным номером високосным. Если год является високосным, то выведите «YES», иначе выведите «NO».<br>
Год является високосным, если его номер кратен 4, но не кратен 100, или если он кратен 400.</p>

<p>Формат входных данных<br>
На вход программе подаётся натуральное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [19]:
a = int(input())
if a % 400 == 0 or (a % 4 == 0 and not a % 100 == 0):
    print("YES")
else:
    print("NO")

YES


### Ход ладьи ♖
<p>Даны две различные клетки шахматной доски. Напишите программу, которая определяет, может ли ладья попасть с первой клетки на вторую одним ходом. Программа получает на вход четыре числа от 1 до 8 каждое, задающие номер столбца и номер строки сначала для первой клетки, потом для второй клетки. Программа должна вывести «YES», если из первой клетки ходом ладьи можно попасть во вторую, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре числа от 1 до 8.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/c13b2aac-577c-45a5-b29e-7136842cb955/" style="height: 200px; width:200px;"/>

<p>Примечание. Шахматная ладья ходит по горизонтали и вертикали.</p>

In [21]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
if a == c or b == d:
    print("YES")
else:
    print("NO")

NO


### Ход короля ♔🌶️
<p>Даны две различные клетки шахматной доски. Напишите программу, которая определяет, может ли король попасть с первой клетки на вторую одним ходом. Программа получает на вход четыре числа от 1 до 8 каждое, задающие номер столбца и номер строки сначала для первой клетки, потом для второй клетки. Программа должна вывести «YES», если из первой клетки ходом короля можно попасть во вторую, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре числа от 1 до 8.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/e759d185-a6d3-47c8-96b2-7b52fa03af09/" style="height: 200px; width:200px;"/>

<p>Примечание. Шахматный король ходит по горизонтали, вертикали и диагонали, но только на 1 клетку.</p>

In [23]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
if -1 <= a - c <= 1 and -1 <= b - d <= 1:
    print("YES")
else:
    print("NO")

NO


## ***Вложенные и каскадные условия***